language/cc/source_groups.go

```diff
@@ -28,6 +28,10 @@ import (
 // groupId represents a unique identifier for a group of source files
 type groupId string
 
+func (id groupId) toRuleName() string {
```

Comment:
I don't think `groupId` should include the path within the repo. It shouldn't go beyond the parent directory name: nothing above that should influence how rules are named.